master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -911,27 +908,32 @@ private[celeborn] class Master(
     // offer slots
     val slots =
       masterSource.sample(MasterSource.OFFER_SLOTS_TIME, s"offerSlots-${Random.nextInt()}") {
-        statusSystem.workers.synchronized {
-          if (slotsAssignPolicy == SlotsAssignPolicy.LOADAWARE) {
-            SlotsAllocator.offerSlotsLoadAware(
-              selectedWorkers,
-              requestSlots.partitionIdList,
-              requestSlots.shouldReplicate,
-              requestSlots.shouldRackAware,
-              slotsAssignLoadAwareDiskGroupNum,
-              slotsAssignLoadAwareDiskGroupGradient,
-              loadAwareFlushTimeWeight,
-              loadAwareFetchTimeWeight,
-              requestSlots.availableStorageTypes)
-          } else {
-            SlotsAllocator.offerSlotsRoundRobin(
-              selectedWorkers,
-              requestSlots.partitionIdList,
-              requestSlots.shouldReplicate,
-              requestSlots.shouldRackAware,
-              requestSlots.availableStorageTypes)
+        statusSystem.synchronizedWorkers(new Supplier[JMap[
```

Comment:
Why is this change necessary? Here needs to do a synchronized operation.

Reply:
yes，because we set the workersMap as private, so I provide a method named `synchronizedWorkers` to lock the workersMap.